Indico/IndicoClient.cs

```diff
@@ -58,9 +58,19 @@ public IndicoClient(IndicoConfig indicoConfig=null)
         /// Create a new GraphQL request
         /// </summary>
         /// <returns>GraphQLRequest</returns>
-        public GraphQLRequest GraphQLRequest()
+        public GraphQLRequest GraphQLRequest(string query=null, string operationName=null)
         {
-            return new GraphQLRequest(this);
+            GraphQLRequest request = new GraphQLRequest(this.GraphQLHttpClient);
+            if (query != null)
+            {
+                request.Query = query;
+            }
+
+            if (operationName != null)
+            {
+                request.OperationName = operationName;
+            }
+            return request;
```

Comment:
If it is necessary to replicate the java API style, This works perfectly, and can be merged.
If not, why don't we let the end user initialize objects directly without the need to call client.function at all.
```
GraphQLRequest request = new GraphQLRequest(client)
 {
    Query = 
    OperationName = 
    Variables = 
 }
```
This will be slightly different from java API where all the objects were part of IndicoClient but will easily adhere to C# style of coding.

This is also prevent us from passing the same parameters again like
```
if (mgId != -1)
{
   mgQuery.MgId = mgId;
}
```
Because since removing the function pattern, there is no simple way to pass the arguments directly. I think we should either keep the functional pattern and client.function style or keep the current getter/setter pattern with object initializers and remove it altogether. I would like to know your thoughts about this.

Reply:
We definitely do not want to remove the IndicoClient class but maybe talk about it in a different way.  Users can always choose to directly create Query or Mutation objects from the classes if they want.

Reply:
thank you for clearing that up.